Solutions/Corvus.UriTemplate.TavisApi.Tests/UriTemplateExtensionsTests.cs

```diff
@@ -252,7 +252,55 @@ public void PartiallyParametersFromAnObjectFromInvalidUrl()
             })
             .Resolve();
 
-            Assert.Equal("http://dev.example.org/v2/customers{?active,country}", url);
+            Assert.Equal("http://dev.example.org/v2/customers{?active}{&country}", url);
+        }
+
+        [Fact]
+        public void PartiallyParametersFromAnObject()
+        {
+
+            var url = new UriTemplate("http://{environment}.example.org/{version}/customers{?active,country}", resolvePartially: true)
+            .AddParameters(new {
+                environment = "dev",
+                version = "v2",
+                active = "foo"
+            })
+            .Resolve();
+
+            Assert.Equal("http://dev.example.org/v2/customers?active=foo{&country}", url);
+        }
+
+        [Fact]
+        public void PartiallyParametersFromAnObjectMissingSecondOfThree()
+        {
+
+            var url = new UriTemplate("http://{environment}.example.org/{version}/customers{?active,country,thing}", resolvePartially: true)
+            .AddParameters(new {
+                environment = "dev",
+                version = "v2",
+                active = "foo",
+                thing = "bar"
+            })
+            .Resolve();
+
+            Assert.Equal("http://dev.example.org/v2/customers?active=foo{&country}&thing=bar", url);
```

Comment:
Wouldn't it be better to have this as `"http://dev.example.org/v2/customers?active=foo&thing=bar{&country}"`?

Reply:
As with the previous comment, in an ideal world, that would be the case; but it would require multiple passes over the string and allocating some kind of look-ahead buffering to achieve that, and we are prioritising perf (providing it remains functionally equivalent).

I will open an enhancement issue for this, though.

Reply:
https://github.com/corvus-dotnet/Corvus.UriTemplates/issues/82